research/2022-11-02-building-privacy-protecting-infrastructure.md

```diff
@@ -0,0 +1,489 @@
+some nodes that do more consumption and other nodes that do more provisioning.
+This gives rise to the idea of a service network, where services are provided
+for and consumed.
+
+![Adaptive Nodes](/img/building_private_infra_adaptive.png)
+
+### Waku - protocol interactions 
+
+There are many protocols that interact. Waku Relay protocol is based libp2p
+GossipSub for p2p messaging. We have filter for bandwidth-restricted nodes to
+only receive subset of messages. Lightpush for nodes with short connection
+windows to push messages into network. Store for nodes that want to retrieve
+historical messages.
+
+On the payload layer, we provide support for Noise handshakes/key-exchanges.
+This means that as a developers, you can get end-to-end encryption and expected
+guarantees out of the box. We have support for setting up a secure channel from
+scratch, and all of this paves the way for providing Signal's Double Ratchet at
+the protocol level much easier. We also have experimental support for
+multi-device usage. Similar features have existed in for example the Status app
+for a while, but with this we make it easier for any platform using Waku to use
+it.
+
+There are other protocols too, related to peer discovery, topic usage, etc. See
+[specs](https://rfc.vac.dev/) for more details.
+
+![Protocol Interactions](/img/building_private_infra_interactions.png)
+
+### Waku - Network
+
+For the Waku network, there are a few problems. For example, when it comes to
+network spam and incentivizing service nodes. We want to address these while
+keeping privacy-guarantees of the base layer. I'm going to go into both of
+these.
+
+The spam problem arises on the gossip layer when anyone can overwhelm the
+network with messages. The service incentivization is a problem when nodes don't
+directly benefit from the provisioning of a certain service. This can happen if
+they are not using the protocol directly themselves as part of normal operation,
+or if they aren't socially inclined to provide a certain service. This depends a
+lot on how an individual platform decides to use the network.
+
+![Waku Network](/img/building_private_infra_network.png)
+
+### Dealing with network spam and RLN Relay
+
+Since the p2p relay network is open to anyone, there is a problem with spam. If
+we look at existing solutions for dealing with spam in traditional messaging
+systems, a lot of entities like Google, Facebook, Twitter, Telegram, Discord use
+phone number verification. While this is largely sybil-resistant, it is
+centralized and not private at all.
+
+Historically, Whisper used PoW which isn't good for heterogenerous networks.
+Peer scoring is open to sybil attacks and doesn't directly address spam
+protection in an anonymous p2p network.
+
+The key idea here is to use RLN for private economic spam protection using
+zkSNARKs.
+
+I'm not going to go into too much detail of RLN here. If you are interested, I
```

Comment:
How about mentioning the published papers as well as the rln and rln-relay specs in here.

Reply:
RLN spec ok